examples/AdminClient/Program.cs

```diff
@@ -757,6 +757,74 @@ static async Task AlterUserScramCredentialsAsync(string bootstrapServers, string
+                await producer.ProduceAsync(TopicName,new Message<Null, string> { Value = "Producer Message", Timestamp = new Timestamp(basetimestamp + 400,TimestampType.CreateTime)});
+                await producer.ProduceAsync(TopicName,new Message<Null, string> { Value = "Producer Message", Timestamp = new Timestamp(basetimestamp + 250,TimestampType.CreateTime)});
+                producer.Flush(new TimeSpan(0,0,10));
+            }
+            var timeout = TimeSpan.FromSeconds(30);
+            ListOffsetsOptions options = new ListOffsetsOptions(){RequestTimeout = timeout, IsolationLevel = Confluent.Kafka.Admin.IsolationLevel.ReadUncommitted};
+            Dictionary<TopicPartition,OffsetSpec> requests = new Dictionary<TopicPartition, OffsetSpec>();
+            TopicPartition tp = new TopicPartition(TopicName,0);
+            using (var adminClient = new AdminClientBuilder(new AdminClientConfig { BootstrapServers = bootstrapServers }).Build())
+            {
+                requests[tp] = OffsetSpec.Earliest;
+                var ListOffsetsResult = await adminClient.ListOffsetsAsync(requests,options);
+                foreach(var ListOffsetsResultInfo in ListOffsetsResult.ListOffsetsResultInfos)
+                {
+                    TopicPartitionOffsetError topic_partition = ListOffsetsResultInfo.TopicPartitionOffsetError;
+                    long Timestamp = ListOffsetsResultInfo.Timestamp;
+                    Console.WriteLine($"{topic_partition.Topic} ${topic_partition.Partition} ${topic_partition.Error.Code} ${topic_partition.Offset} ${Timestamp}");
+                }
+
+                requests[tp] = OffsetSpec.Latest;
+                ListOffsetsResult = await adminClient.ListOffsetsAsync(requests,options);
+                foreach(var ListOffsetsResultInfo in ListOffsetsResult.ListOffsetsResultInfos)
+                {
+                    TopicPartitionOffsetError topic_partition = ListOffsetsResultInfo.TopicPartitionOffsetError;
+                    long Timestamp = ListOffsetsResultInfo.Timestamp;
+                    Console.WriteLine($"{topic_partition.Topic} ${topic_partition.Partition} ${topic_partition.Error.Code} ${topic_partition.Offset} ${Timestamp}");
+                }
+
+                requests[tp] = OffsetSpec.MaxTimestamp;
+                ListOffsetsResult = await adminClient.ListOffsetsAsync(requests,options);
+                foreach(var ListOffsetsResultInfo in ListOffsetsResult.ListOffsetsResultInfos)
+                {
+                    TopicPartitionOffsetError topic_partition = ListOffsetsResultInfo.TopicPartitionOffsetError;
+                    long Timestamp = ListOffsetsResultInfo.Timestamp;
+                    Console.WriteLine($"{topic_partition.Topic} ${topic_partition.Partition} ${topic_partition.Error.Code} ${topic_partition.Offset} ${Timestamp}");
+                }
```

Comment:
Needs to be updated with the current API. For each of the OffsetSpec also print what offset spec is being used.